### PR TITLE
feat: Add form data provider to speed up forms

### PR DESCRIPTION
**Pull request by yaraslau-kavaliou**
Ticket - https://oat-sa.atlassian.net/browse/REL-1323

Related PR - https://github.com/oat-sa/tao-core/pull/3929

### CODE REVIEW

File: core/kernel/persistence/starsql/helper/RecordProcessor.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        return $filteredValues;
+    }
+
+    public function filterRecordsByAvailableLanguage($entries, $dataLanguage, $defaultLanguage): array
+    {
+        $fallbackLanguage = '';
+
+        foreach ($entries as $entry) {
```
Comment (vbyndych) [inline core/kernel/persistence/starsql/helper/RecordProcessor.php:48]: This line throws following error on kitchen instance:
```
2023-12-04 15:35:07 [ERROR] [tao] 'php error(2): foreach() argument must be of type array|object, null given' /tao/code/generis/core/kernel/persistence/starsql/helper/RecordProcessor.php 48

```

**Review by augustas — COMMENTED** [review]
Some comments, but overall pretty nice

**Review by augustas — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: core/kernel/persistence/starsql/helper/RecordProcessor.php (hunk 1 above)
Comment (yaraslau-kavaliou) [reply core/kernel/persistence/starsql/helper/RecordProcessor.php]: This have been fixed

**Pull request merged**